src/hdmf/container.py

```diff
@@ -85,10 +86,78 @@ def setter(self, val):
+        """
+        This method will be called in the setter. The termset configuration will be used (if loaded)
+        to check for a defined TermSet associated with the field. If found, the value of the field
+        will be wrapped with a TermSetWrapper.
+
+        Even though the path field in the configurator can be a list of paths, the config
+        itself is only one file. When a user loads custom configs, the config is appended/modified.
+        The modifications are not written to file, avoiding permanent modifications.
+        """
+        # load termset configuration file from global Config
+        try:
+            configurator = self.type_map.ts_config
+            if len(configurator.path)>0:
+                # The type_map has a config always set; however, when toggled off, the config path is empty.
+                CUR_DIR = os.path.dirname(os.path.realpath(configurator.path[0]))
+                termset_config = configurator.config
+            else:
+                return val
+        except AttributeError:
+            # These are for classes that have not been registered, e.g., some test containers.
+            # Without register_class, the type_map parameter is not defined.
+            return val
+
+        # check to see that the namespace for the container is in the config
+        if self.namespace not in self.type_map.container_types:
+            msg = "%s not found within loaded configuration." % self.namespace
+            warn(msg)
+            return val
+        else:
+            # check to see that the container type is in the config under the namespace
+            config_namespace = termset_config['namespaces'][self.namespace]
+            object_name = self.__class__.__name__
+
+            if object_name not in config_namespace['data_types']:
+                msg = '%s not found within the configuration for %s' % (object_name, self.namespace)
+                warn(msg)
+                return val
+            else:
+                for attr in config_namespace['data_types'][object_name]:
+                    obj_mapper = self.type_map.get_map(self)
+
+                    # get the spec according to attr name in schema
+                    # Note: this is the name for the field in the config
+                    spec = obj_mapper.get_attr_spec(attr)
+
+                    # In the case of dealing with datasets directly or not defined in the spec.
+                    # (Data/VectorData/DynamicTable/etc)
+                    if spec is None:
+                        msg = "Spec not found for %s." % attr
+                        warn(msg)
+                        return val
+                    else:
+                        # From the spec, get the corresponding constructor name
+                        constr_name = obj_mapper.get_const_arg(spec)
+                        termset_path = os.path.join(CUR_DIR, config_namespace['data_types'][object_name][constr_name])
```

Comment:
```suggestion
                        # From the spec, get the mapped attribute name
                        mapped_attribute_name = obj_mapper.get_attribute(spec)
                        termset_path = os.path.join(CUR_DIR, config_namespace['data_types'][object_name][mapped_attribute_name])
```

It looks odd to see the constructor argument name being used here. In 100% of custom object mappings that I am aware of, the mapped constructor arg and the mapped attribute name are the same, so it does not matter, but in the event of a discrepancy, it seems better to use attribute names in the config file than constructor arg names.